### DISCUSSION TITLE
Web Test For Razor Pages In Tired Project

### DISCUSSION THREAD

**Discussion by mostafa1790** [abpframework/abp#9398] [Q&A]
Hi,

Anyone tried to make Web Test For Razor Pages In Tired Project.

Thanks

**Comment by realLiangshiwei** [comment]
The tiered project needs to start the `HttpApi.Host` and `IdentityServer` projects, it's hard for `Web.Tests`.

**Reply by mostafa1790** [reply]
thanks for your reply but can you tell me a best practice to test razor pages or a work around in abp
tired project

**Reply by mostafa1790** [reply]
@realLiangshiwei 

Kindly can you help me if there are an alternative way to test razor pages

Thanks

**Reply by d-oit** [reply]
@mostafa1790  Try https://testproject.io/ or https://www.katalon.com/ for UI tests.
